rcl/src/rcl/publisher.c

```diff
@@ -297,6 +297,7 @@ rcl_publish_serialized_message(
     return RCL_RET_PUBLISHER_INVALID;  // error already set
   }
   RCL_CHECK_ARGUMENT_FOR_NULL(serialized_message, RCL_RET_INVALID_ARGUMENT);
+  TRACETOOLS_TRACEPOINT(rcl_publish, (const void *)publisher, (const void *)serialized_message);
```

Comment:
Shouldn't you also instrument the serialized version of take? In `rcl_take_serialized_message()`.

Reply:
That's correct. Fixed in https://github.com/ros2/rcl/pull/1136/commits/e0ed710b7356abf216869f7f9b1c08e45c5787c5.